src/App.tsx

```diff
@@ -1,7 +1,42 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+type State = {
+  pressedKey: string;
+  wasSomethingPressed: boolean;
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    pressedKey: 'Nothing was pressed yet',
+    wasSomethingPressed: false,
+  };
+
+  componentDidMount(): void {
+    document.addEventListener('keyup', this.handleKeyUp);
+  }
+
+  componentWillUnmount(): void {
+    document.removeEventListener('keyup', this.handleKeyUp);
+  }
+
+  handleKeyUp = (event: KeyboardEvent): void => {
+    this.setState({
+      pressedKey: event.key,
+      wasSomethingPressed: true,
+    });
+  };
+
+  render() {
+    const { pressedKey, wasSomethingPressed } = this.state;
+
+    return (
+      <div className="App">
+        <p className="App__message">
+          {wasSomethingPressed
```

Comment:
You can simplify the message rendering by checking if 'pressedKey' is empty or has the initial value like: 'pressedKey === '' || pressedKey === 'No key has been pressed yet' ? 'No key has been pressed yet' : `The last pressed key is [${pressedKey}]` 🚀